### PR TITLE
[AMORO-2592] Optimizing location about building steps so that contributors fo find it easier

### LINKED ISSUE #2592: [Improvement]: Optimizing the doc for building steps
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Currently the building steps locate in README, but people who contribute to the project always search in the CONTRIBUTING doc

### How should we improve?

add Building steps in CONTRIBUTING 

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by klion26**
## Why are the changes needed?

Close #2592 .

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,40 +108,6 @@ Amoro contains modules as below:
     - `amoro-spark` provides Spark connectors for Mixed format tables (use amoro-spark-runtime for a shaded version)
     - `amoro-trino` provides Trino connectors for Mixed format tables
 
-
```
Comment (klion26) [inline README.md:111]: IMHO, people who find out the project first time are more interested in how to use it than in building it, so I move the build steps directly to `CONTRIBUTING`

**Comment by klion26** [comment]
cc @zhoujinsong

File: README.md (hunk 1 above)
Comment (zhoujinsong) [reply README.md:111]: GitHub is the platform for developers.
So I prefer keeping the building sections int README for developers.
And I found that some other open-source projects closely related to Amoro are also doing the same，like:

- Apache Iceberg: https://github.com/apache/iceberg/blob/main/README.md
- Apache Flink: https://github.com/apache/flink/blob/main/README.md
- Apache Paimon(incubating): https://github.com/apache/incubator-paimon/blob/main/README.md

Comment (klion26) [reply README.md:111]: ok，will keep the build steps in README

Comment (klion26) [reply README.md:111]: ok，will keep the build steps in README

**Review by zhoujinsong — APPROVED** [review]
LGTM.
Thanks a lot for your contribution!

**Comment by klion26** [comment]
@zhoujinsong thanks for the review and merging!

**Pull request merged**